drivers/sdhc/sdhci_common.c

```diff
@@ -428,6 +429,71 @@ static int sdhci_transfer_data(struct sdhci_common *sdhci_ctx, struct sdhc_data
 	return sdhci_poll_data_complete(sdhci_ctx, timeout / 100);
 }
 
+static void sdhc_adma_desc_fill(struct sdhc_adma_desc *desc, uintptr_t dma_addr, uint16_t len,
+				bool end)
+{
+	uint16_t cmd;
+
+	cmd = SDHC_ADMA_DESC_VALID | SDHC_ADMA_DESC_ACT_TRAN;
+	if (end) {
+		cmd |= SDHC_ADMA_DESC_END;
+	}
+
+	desc->cmd = sys_cpu_to_le16(cmd);
+	desc->len = sys_cpu_to_le16(len);
+	desc->addr = sys_cpu_to_le32(dma_addr);
+	LOG_DBG("sdhc:dma_desc: %016llx", *((uint64_t *)desc));
+}
+
+static void sdhc_prepare_adma_table(struct sdhci_common *sdhci_ctx, struct sdhc_data *data,
+				    uintptr_t dma_addr)
+{
+	uint32_t trans_bytes = data->block_size * data->blocks;
+	struct sdhc_adma_desc *desc = sdhci_ctx->adma_descs;
+	uintptr_t desc_dma_addr;
+	uint32_t desc_count;
+
+	desc_count = DIV_ROUND_UP(trans_bytes, SDHC_ADMA_DESC_MAX_LEN);
+	desc_dma_addr = z_mem_phys_addr(sdhci_ctx->adma_descs);
+	while (--desc_count) {
+		sdhc_adma_desc_fill(desc, dma_addr, SDHC_ADMA_DESC_MAX_LEN, false);
+		dma_addr += SDHC_ADMA_DESC_MAX_LEN;
+		trans_bytes -= SDHC_ADMA_DESC_MAX_LEN;
+		desc++;
+	}
+
+	sdhc_adma_desc_fill(desc, dma_addr, trans_bytes, true);
+
+#if !defined(CONFIG_NOCACHE_MEMORY)
+	sys_cache_data_flush_range(sdhci_ctx->adma_descs,
+				   (sdhci_ctx->adma_descs_num * sizeof(struct sdhc_adma_desc)));
+#endif
+
+	sys_write32(desc_dma_addr, sdhci_ctx->reg_base + SDHCI_ADMA_ADDRESS);
+}
+
+static void sdhci_set_dma_sel(struct sdhci_common *sdhci_ctx)
+{
+	uint8_t ctrl;
+
+	if (!sdhci_ctx->f_use_dma) {
+		return;
+	}
+
+	/* Set DMA Select */
+	ctrl = sys_read8(sdhci_ctx->reg_base + SDHCI_HOST_CONTROL);
+
+	ctrl &= ~SDHCI_CTRL_DMA_MASK;
+	if (sdhci_ctx->dma_mode == SDHC_DMA_SDMA) {
+		ctrl |= SDHCI_CTRL_SDMA << SDHCI_CTRL_DMA_SHIFT;
+	} else if (sdhci_ctx->dma_mode == SDHC_DMA_ADMA2) {
+		ctrl |= SDHCI_CTRL_ADMA2 << SDHCI_CTRL_DMA_SHIFT;
+	}
+
+	LOG_DBG("sdhc:dma_sel: SDHCI_HOST_CONTROL:%02x", ctrl);
```

Comment:
IIRC macros LOG_DBG and LOG_ERR can add "sdhc" to the output if the configured to do so. If I'm right then "sdhc:" should be removed

Reply:
It depends on LOG cfg - for example with CONFIG_LOG_MODE_MINIMAL it will not do such things..

Reply:
That's why I think "sdhc" mention is excessive

Reply:
The current log looks like below:
```
I: sdhc: detected ver:2 vendor_ver:16 supported
I: sdhc: capabilities caps0:15eac832 caps1:8000a577
I: mmc@1000fff000:"reset max bus frequency to 200000000 from 0"
```
if I remove "sdhc" and there will be more logs (DEBUG level) it will be impossible to determine which one belongs to SDHC. So, I've had to add it while developing and debugging to have logs readable.

Reply:
Please see link: https://docs.zephyrproject.org/latest/services/logging/index.html#logging-in-a-module
```
[00:00:00.000,274] <info> sample_instance.inst1: logging message
```
This example was taken from there

Reply:
It is used in all LOGS is this module. If you insist - ok, The "sdhc" prefix can be removed, but logically it should be done as separate PR all at once and not as part of this one to avoid merge conflicts with pending PRs. 

So, if you wish to continue, I think right approach would be create todo/task and solve at once functional PRs are complited.